graylog2-server/src/main/java/org/graylog/plugins/sidecar/services/ConfigurationService.java

```diff
@@ -192,6 +193,10 @@ private String renderTemplate(String configurationId, Map<String, Object> contex
             throw new RenderTemplateException(e.getMessage(), e);
         }
 
-        return writer.toString();
+        template = writer.toString();
+        if (!template.substring(template.length() -1).equals("\n")) {
```

Comment:
I think we can simplify the condition with `!template.endsWith("\n")`.

I also think it's clearer adding the conditional into the return sentence (but this is more of a personal thing, so please disregard if you don't agree):

`return template.endsWith("\n") ? template : template + "\n";`